autorest/codegen/serializers/__init__.py

```diff
@@ -88,31 +88,33 @@ def _serialize_and_write_package_files_proc(**kwargs: Any):
                 self._autorestapi.write_file(out_path / output_name, render_result)
 
         def _prepare_params() -> Dict[Any, Any]:
-            package_parts = code_model.options["package_name"].split("-")[:-1]
+            package_parts = self.code_model.options["package_name"].split("-")[:-1]
+            token_cred = isinstance(self.code_model.credential_schema_policy.credential, TokenCredentialSchema)
             params = {
-                "release_time": time.strftime('%Y-%m-%d', time.localtime()),
-                "nspkg_names": [".".join(package_parts[: i + 1]) for i in range(len(package_parts))],
+                "token_credential": token_cred,
+                "pkgutil_names": [".".join(package_parts[: i + 1]) for i in range(len(package_parts))],
                 "init_names": ["/".join(package_parts[: i + 1]) + "/__init__.py" for i in range(len(package_parts))]
             }
-            params.update(code_model.options)
+            params.update(self.code_model.options)
+            params.update(self.code_model.package_dependency)
             return params
 
-        count = code_model.options["package_name"].count("-") + 1
+        count = self.code_model.options["package_name"].count("-") + 1
         for _ in range(count):
             out_path = out_path / Path("..")
 
-        if code_model.options["package_mode"] in ("dataplane", "mgmtplane"):
+        if self.code_model.options["package_mode"] in ("dataplane", "mgmtplane"):
             env = Environment(
                 loader=PackageLoader("autorest.codegen", "templates/templates_package"),
                 undefined=StrictUndefined)
             _serialize_and_write_package_files_proc(**_prepare_params())
-        elif Path(code_model.options["package_mode"]).exists():
+        elif Path(self.code_model.options["package_mode"]).exists():
```

Comment:
I think it's a little weird to do Path().exists(), can you just check that package_mode is not None?

Reply:
None check has been done before stepping into the function. `.exists()` is to avoid customers input invalid folder carelessly.